File: fix-all-jsx-final.js
#!/usr/bin/env node

/**
 * Script FINAL pour corriger TOUTES les balises JSX non fermées en une seule passe
 */

const fs = require('fs');
const path = require('path');

const FILES = [
  'apps/frontend/src/app/(dashboard)/dashboard/customize/page.tsx',
  'apps/frontend/src/app/(dashboard)/dashboard/integrations/page.tsx',
  'apps/frontend/src/app/(dashboard)/dashboard/library/page.tsx',
  'apps/frontend/src/app/(dashboard)/dashboard/monitoring/page.tsx',
  'apps/frontend/src/app/(dashboard)/dashboard/orders/page.tsx',
];

function fixFile(filePath) {
  console.log(`\n🔧 Correction de: ${filePath}`);
  
  if (!fs.existsSync(filePath)) {
    console.log(`❌ Fichier non trouvé`);
    return { fixed: false };
  }

  let content = fs.readFileSync(filePath, 'utf8');
  const originalContent = content;
  let fixed = false;

  // 1. Corriger les balises <p> non fermées
  content = content.replace(/<p([^>]*)>([^<]*?)(?=\s*<\/div>|\s*<\/Card>|\s*<\/CardContent>|\s*<\/span>|\s*<\/h3>|\s*<\/h4>)/g, (match, attrs, inner) => {
    if (!match.includes('</p>')) {
      fixed = true;
      return `<p${attrs}>${inner}</p>`;
    }
    return match;
  });

  // 2. Corriger les balises <span> non fermées
  content = content.replace(/<span([^>]*)>([^<]*?)(?=\s*<\/div>|\s*<\/Button>|\s*<\/p>)/g, (match, attrs, inner) => {
    if (!match.includes('</span>')) {
      fixed = true;
      return `<span${attrs}>${inner}</span>`;
    }
    return match;
  });

  // 3. Corriger les balises <TabsTrigger> non fermées
  content = content.replace(/<TabsTrigger([^>]*)>([^<]+?)(?=\s*<TabsTrigger|\s*<\/TabsList>)/g, (match, attrs, inner) => {
    if (!match.includes('</TabsTrigger>')) {
      fixed = true;
      return `<TabsTrigger${attrs}>${inner}</TabsTrigger>`;
    }
    return match;
  });

  // 4. Corriger les balises <SelectItem> non fermées
  content = content.replace(/<SelectItem([^>]*)>([^<]+?)(?=\s*<SelectItem|\s*<\/SelectContent>)/g, (match, attrs, inner) => {
    if (!match.includes('</SelectItem>')) {
      fixed = true;
      return `<SelectItem${attrs}>${inner}</SelectItem>`;
    }
    return match;
  });

  // 5. Corriger les balises <Button> non fermées avant </div>
  content = content.replace(/<Button([^>]*)>([^<]*?)(?=\s*<\/div>|\s*<\/DialogFooter>|\s*<\/DialogTrigger>)/g, (match, attrs, inner) => {
    if (!match.includes('</Button>')) {
      fixed = true;
      return `<Button${attrs}>${inner}</Button>`;
    }
    return match;
  });

  // 6. Corriger les structures avec </div> manquant avant </>
  content = content.replace(/<\/>\s*<\/div>/g, '</div>\n                </>');
  content = content.replace(/<\/>\s*\)\s*\)/g, '</div>\n                </>');

  // 7. Corriger les </Card> manquants avant ))
  content = content.replace(/<\/div>\s*<\/div>\s*\)\s*\)/g, (match) => {
    if (!match.includes('</Card>')) {
      fixed = true;
      return '</div>\n            </div>\n          </Card>\n          ));';
    }
    return match;
  });

  // 8. Corriger les balises </motion.div> manquantes
  content = content.replace(/<\/Card>\s*\)\s*\)/g, '</Card>\n        </motion.div>\n      );');

  // 9. Corriger les structures avec </div> manquant
  content = content.replace(/<\/span>\s*<\/span>/g, '</span>');

  if (fixed && content !== originalContent) {
    fs.writeFileSync(filePath, content);
    console.log(`✅ Fichier corrigé`);
  } else {
    console.log(`✓ Aucune correction nécessaire`);
  }

  return { fixed };
}

function main() {
  console.log('🚀 Correction FINALE de toutes les balises JSX...\n');
  
  const workspaceRoot = process.cwd();
  let totalFixed = 0;

  for (const filePath of FILES) {
    const fullPath = path.join(workspaceRoot, filePath);
    const result = fixFile(fullPath);
    if (result.fixed) {
      totalFixed++;
    }
  }

  console.log(`\n📊 Résumé: ${totalFixed} fichiers corrigés`);
}

if (require.main === module) {
  main();
}

module.exports = { fixFile };

File: scripts/fix-all-jsx-final.py
#!/usr/bin/env python3
"""
Script final de correction de toutes les erreurs JSX
Corrige systématiquement tous les problèmes identifiés
"""

import re
import os
from pathlib import Path

DASHBOARD_DIR = Path(__file__).parent.parent / 'apps/frontend/src/app/(dashboard)/dashboard'

# Composants auto-fermants
SELF_CLOSING = ['Separator', 'Progress', 'Checkbox', 'Input', 'Textarea']

def fix_file(file_path):
    """Corrige toutes les erreurs JSX dans un fichier"""
    with open(file_path, 'r', encoding='utf-8') as f:
        lines = f.readlines()
    
    original_lines = lines.copy()
    modified = False
    
    # 1. Supprimer fermetures pour composants auto-fermants
    for i, line in enumerate(lines):
        for comp in SELF_CLOSING:
            if f'</{comp}>' in line:
                lines[i] = line.replace(f'</{comp}>', '')
                modified = True
    
    # 2. Corriger Buttons non fermés
    for i in range(len(lines)):
        line = lines[i]
        if '<Button' in line and '</Button>' not in line and '/>' not in line:
            # Chercher la fermeture dans les lignes suivantes
            found_close = False
            for j in range(i + 1, min(i + 10, len(lines))):
                if '</Button>' in lines[j]:
                    found_close = True
                    break
                # Si on trouve </div>, </CardContent>, </Card> avant </Button>, c'est une erreur
                if '</div>' in lines[j] or '</CardContent>' in lines[j] or '</Card>' in lines[j]:
                    # Ajouter </Button> avant cette ligne
                    indent = len(line) - len(line.lstrip())
                    lines.insert(j, ' ' * indent + '</Button>\n')
                    modified = True
                    found_close = True
                    break
    
    # 3. Corriger Badges non fermés
    for i in range(len(lines)):
        line = lines[i]
        if '<Badge' in line and '</Badge>' not in line and '/>' not in line:
            found_close = False
            for j in range(i + 1, min(i + 10, len(lines))):
                if '</Badge>' in lines[j]:
                    found_close = True
                    break
                if '</div>' in lines[j] or '</CardContent>' in lines[j]:
                    indent = len(line) - len(line.lstrip())
                    lines.insert(j, ' ' * indent + '</Badge>\n')
                    modified = True
                    found_close = True
                    break
    
    # 4. Nettoyer les caractères spéciaux suspects
    for i in range(len(lines)):
        # Supprimer les caractères de contrôle non-ASCII sauf les sauts de ligne
        cleaned = ''.join(c if (32 <= ord(c) <= 126) or c in '\n\r\t' else ' ' for c in lines[i])
        if cleaned != lines[i]:
            lines[i] = cleaned
            modified = True
    
    if modified and lines != original_lines:
        with open(file_path, 'w', encoding='utf-8') as f:
            f.writelines(lines)
        return True
    
    return False

def main():
    print('🔧 Correction finale de toutes les erreurs JSX...\n')
    
    # Fichiers avec erreurs critiques
    critical_files = [
        'ab-testing/page.tsx',
        'affiliate/page.tsx',
        'ai-studio/animations/page.tsx',
        'ai-studio/templates/page.tsx',
        'ai-studio/3d/page.tsx',
        'integrations/page.tsx',
        'ar-studio/preview/page.tsx',
        'analytics/page.tsx'
    ]
    
    fixed_count = 0
    
    for file in critical_files:
        file_path = DASHBOARD_DIR / file
        if file_path.exists():
            if fix_file(file_path):
                fixed_count += 1
                print(f'✅ {file}')
            else:
                print(f'⏭️  {file} (pas de corrections)')
        else:
            print(f'⚠️  {file} (non trouvé)')
    
    print(f'\n📊 {fixed_count} fichier(s) corrigé(s)\n')

if __name__ == '__main__':
    main()











